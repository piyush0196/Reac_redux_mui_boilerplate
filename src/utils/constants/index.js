import { EnglishIcon } from 'components/ui-kit/icons/iconComponents/EnglishIcon'
import { FrenchIcon } from 'components/ui-kit/icons/iconComponents/FrenchIcon'
import { GermanIcon } from 'components/ui-kit/icons/iconComponents/GermanIcon'
import { PortugalIcon } from 'components/ui-kit/icons/iconComponents/PortugalIcon'
import { RussianIcon } from 'components/ui-kit/icons/iconComponents/RussianIcon'
import { SpanishIcon } from 'components/ui-kit/icons/iconComponents/SpanishIcon'

/* ==========================================================================
  LANGUAGES
  ========================================================================== */
const LANGUAGES = [
  { label: 'English', languageCode: 'en', flag: EnglishIcon },
  { label: 'German', languageCode: 'de', flag: GermanIcon },
  { label: 'Spanish', languageCode: 'es', flag: SpanishIcon },
  { label: 'French', languageCode: 'fr', flag: FrenchIcon },
  { label: 'Portuguese', languageCode: 'pt', flag: PortugalIcon },
  { label: 'Russian', languageCode: 'ru', flag: RussianIcon }
]

/* ==========================================================================
  HTTP Method Types
========================================================================== */
const METHOD_TYPES = {
  get: 'get',
  post: 'post',
  put: 'put',
  delete: 'delete',
  patch: 'patch'
}

/* ==========================================================================
LocalStorage / Cookie data
========================================================================== */
const TOKEN = 'authtoken'
const LANGUAGE = 'language'

/* ==========================================================================
Loader types
========================================================================== */
const LOADER_TYPE = {
  SCALE: 'scale',
  PULSE: 'pulse'
}

/* ==========================================================================
  TOASTER / NOTIFICATION
========================================================================== */
const TOASTER_TYPE = {
  success: 'success',
  error: 'error',
  warning: 'warning',
  info: 'info'
}

/* ==========================================================================
  Loader Type Array
========================================================================== */
const LOADER_HANDLER_TYPES = {
  page: 'pageLoader',
  submit: 'submitButtonLoader',
  table: 'tableLoader'
}

/* ==========================================================================
  All the navigation route Paths
========================================================================== */
const gamesSectionParentRoute = '/games'

const ROUTE_PATHS = {
  // ----single routes------------
  rootPath: '/',
  login: '/login',
  settings: '/settings',
  dashboard: '/dashboard',
  profile: '/profile',
  changePassword: '/change-password',
  forgotPassword: '/forgot-password',
  // ----games section parent child routes------------
  games: gamesSectionParentRoute,
  perRoundReportReports: `${gamesSectionParentRoute}/pre-round-reports`,
  cancelledGames: `${gamesSectionParentRoute}/cancelled-games`
}

export {
  TOASTER_TYPE,
  LANGUAGES,
  TOKEN,
  LANGUAGE,
  ROUTE_PATHS,
  LOADER_TYPE,
  METHOD_TYPES,
  LOADER_HANDLER_TYPES
}
