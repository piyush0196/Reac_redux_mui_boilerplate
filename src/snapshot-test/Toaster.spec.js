// import TestRenderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { TOASTER_TYPE } from 'utils/constants'
import { Toaster } from 'components/ui-kit/AppToaster'

describe('Component - Toaster', () => {
  const wrapper1 = shallow(
    <Toaster
      toasterType={TOASTER_TYPE.success}
      openToaster
      toasterMessage='Default Message'
      handleToasterClose={() => {}}
    />
  )

  const wrapper2 = shallow(
    <Toaster
      toasterType={TOASTER_TYPE.error}
      openToaster
      toasterMessage='Default Message'
      handleToasterClose={() => {}}
    />
  )

  const wrapper3 = shallow(
    <Toaster
      toasterType={TOASTER_TYPE.warning}
      openToaster
      toasterMessage='Default Message'
      handleToasterClose={() => {}}
    />
  )

  const wrapper4 = shallow(
    <Toaster
      toasterType={TOASTER_TYPE.info}
      openToaster
      toasterMessage='Default Message'
      handleToasterClose={() => {}}
    />
  )

  it('renders Success Toaster correctly', () => {
    expect(shallowToJson(wrapper1)).toMatchSnapshot()
  })

  it('renders Error Toaster correctly', () => {
    expect(shallowToJson(wrapper2)).toMatchSnapshot()
  })

  it('renders Warning Toaster correctly', () => {
    expect(shallowToJson(wrapper3)).toMatchSnapshot()
  })

  it('renders Info Toaster correctly', () => {
    expect(shallowToJson(wrapper4)).toMatchSnapshot()
  })
})
