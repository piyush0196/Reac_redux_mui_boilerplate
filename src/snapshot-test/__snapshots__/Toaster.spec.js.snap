// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Toaster renders Error Toaster correctly 1`] = `
<ForwardRef(Snackbar)
  TransitionComponent={[Function]}
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={3000}
  onClose={[Function]}
  open={true}
  style={
    Object {
      "maxWidth": "95vw",
      "padding": 0,
      "width": "fit-content",
    }
  }
>
  <ForwardRef(Alert)
    onClose={[Function]}
    severity="error"
    style={
      Object {
        "fontSize": 16,
        "opacity": 0.95,
      }
    }
    variant="filled"
  >
    Default Message
  </ForwardRef(Alert)>
</ForwardRef(Snackbar)>
`;

exports[`Component - Toaster renders Info Toaster correctly 1`] = `
<ForwardRef(Snackbar)
  TransitionComponent={[Function]}
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={3000}
  onClose={[Function]}
  open={true}
  style={
    Object {
      "maxWidth": "95vw",
      "padding": 0,
      "width": "fit-content",
    }
  }
>
  <ForwardRef(Alert)
    onClose={[Function]}
    severity="info"
    style={
      Object {
        "fontSize": 16,
        "opacity": 0.95,
      }
    }
    variant="filled"
  >
    Default Message
  </ForwardRef(Alert)>
</ForwardRef(Snackbar)>
`;

exports[`Component - Toaster renders Success Toaster correctly 1`] = `
<ForwardRef(Snackbar)
  TransitionComponent={[Function]}
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={3000}
  onClose={[Function]}
  open={true}
  style={
    Object {
      "maxWidth": "95vw",
      "padding": 0,
      "width": "fit-content",
    }
  }
>
  <ForwardRef(Alert)
    onClose={[Function]}
    severity="success"
    style={
      Object {
        "fontSize": 16,
        "opacity": 0.95,
      }
    }
    variant="filled"
  >
    Default Message
  </ForwardRef(Alert)>
</ForwardRef(Snackbar)>
`;

exports[`Component - Toaster renders Warning Toaster correctly 1`] = `
<ForwardRef(Snackbar)
  TransitionComponent={[Function]}
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={3000}
  onClose={[Function]}
  open={true}
  style={
    Object {
      "maxWidth": "95vw",
      "padding": 0,
      "width": "fit-content",
    }
  }
>
  <ForwardRef(Alert)
    onClose={[Function]}
    severity="warning"
    style={
      Object {
        "fontSize": 16,
        "opacity": 0.95,
      }
    }
    variant="filled"
  >
    Default Message
  </ForwardRef(Alert)>
</ForwardRef(Snackbar)>
`;
